packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -278,12 +278,28 @@ class FileNode extends TreeNode<FileNode> {
     final root = FileNode('<root>');
 
     for (var script in scripts) {
-      var parts = script.uri.split('/');
+      final uri = Uri.parse(script.uri);
+      final scheme = uri.scheme;
+      var parts = uri.path.split('/');
 
-      // Look for and handle dotted package names.
-      if (parts.isNotEmpty && parts.first.contains('.')) {
+      // handle google3:///foo/bar
+      if (parts.first.isEmpty) {
+        parts = parts.where((part) => part.isNotEmpty).toList();
         parts = [
-          ...parts.first.split('.'),
+          '$scheme:${parts.first}',
+          ...parts.sublist(1),
```

Comment:
Why the sublist here?

Reply:
Here, we're concatenating the first part of the file path to the schema. This is so that user's see 'google3:foo' as the top level entry in the scripts list instead of `google3` (w/ lots of entries below it).

I'll add a comment. I may also pull this logic (parsing a uri) out into a new method, separate from the code to create a tree structure from the scripts.

Reply:
I added a comment about why this is here (and, refactoring the path segment munging into a separate class for clarity).